vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileViewModel.kt

```diff
@@ -112,6 +117,36 @@ class RoomProfileViewModel @AssistedInject constructor(
                 }
     }
 
+    private fun observeAdminMembers() {
+        val roomMemberQueryParams = roomMemberQueryParams {
+            displayName = QueryStringValue.IsNotEmpty
+            memberships = Membership.activeMemberships()
+        }
+
+        combine(
+                room.flow().liveRoomMembers(roomMemberQueryParams),
+                room.flow()
+                        .liveStateEvent(EventType.STATE_ROOM_POWER_LEVELS, QueryStringValue.NoCondition)
+                        .mapOptional { it.content.toModel<PowerLevelsContent>() }
+                        .unwrap()
+        ) { roomMembers, powerLevelsContent ->
+            buildAdminMembersList(powerLevelsContent, roomMembers)
+        }
+                .execute { async ->
+                    copy(isLastAdmin = async.invoke()?.size == 1 &&
+                            async.invoke()?.find { roomMember ->
+                                roomMember.userId == session.myUserId
+                            } != null)
+                }
+    }
+
+    private fun buildAdminMembersList(powerLevelsContent: PowerLevelsContent, roomMembers: List<RoomMemberSummary>): List<RoomMemberSummary> {
+        return roomMembers
+                .mapNotNull { roomMember ->
+                    if (PowerLevelsHelper(powerLevelsContent).getUserRole(roomMember.userId) == Role.Admin) roomMember else null
+                }
```

Comment:
I think the `filter` operator does the trick:

```suggestion
        return roomMembers.filter { PowerLevelsHelper(powerLevelsContent).getUserRole(it.userId) == Role.Admin) }
```